src/components/providers/PushNotificationsProvider.tsx

```diff
@@ -0,0 +1,101 @@
+import React, { createContext, useContext, useEffect, useState } from "react";
+import { Alert, Platform } from "react-native";
+import * as Notifications from "expo-notifications";
+import * as Device from "expo-device";
+
+type PushNotificationsContextState = {
+  isPushEnabled: boolean;
+  setPushEnabled?: (value: boolean) => void;
+  sendPushNotification?: (notificationContent: {
+    title: string;
+    body: string;
+  }) => Promise<void>;
+};
+
+const initialState: PushNotificationsContextState = {
+  isPushEnabled: false,
+};
+
+const PushNotificationsContext = createContext<PushNotificationsContextState>(initialState);
+
+const PushNotificationsProvider: React.FC<{ children: React.ReactNode }> = ({ children }) => {
+  const [isPushEnabled, setPushEnabled] = useState<boolean>(false);
+
+  useEffect(() => {
+    registerForPushNotificationsAsync();
+  }, []);
+
+  const registerForPushNotificationsAsync = async () => {
+    if (Device.isDevice) {
+      const { status: existingStatus } = await Notifications.getPermissionsAsync();
+      let finalStatus = existingStatus;
+
+      if (existingStatus !== 'granted') {
+        const { status } = await Notifications.requestPermissionsAsync();
+        finalStatus = status;
+      }
+
+      if (finalStatus !== 'granted') {
+        Alert.alert('Failed to get push token for push notification!');
+        return;
+      }
+
+      const token = (await Notifications.getExpoPushTokenAsync()).data;
+      console.log('Expo Push Token:', token);
```

Comment:
remove console log

Reply:
what is this token for?

Reply:
This token gets the id of the phone. Is to ensure you are using a real phone and not a emulator. In the other hand we can use it to create the tempo account of the users